webapp/src/simulator.ts

```diff
@@ -235,7 +235,7 @@ export function makeDirty() { // running outdated code
     dirty = true;
 
     // No need to continue running if we're dirty
-    driver.suspend();
+    if (driver.state == pxsim.SimulatorState.Running) driver.suspend();
```

Comment:
This should be handled inside the driver class.

Reply:
I think this is the right place for this check, the simdriver should be able to switch to the suspended state from any other state at the demand of the editor. However, in this case, the editor is invalidating the code (change has been made) and we only want to suspend the simulator if it's running. ie: don't want to go from Stopped -> Suspended just because we invalidated code.